src/main/java/io/pravega/connectors/flink/FlinkPravegaInputFormat.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.flink;
+
+import com.google.common.base.Preconditions;
+
+import io.pravega.client.admin.ReaderGroupManager;
+import io.pravega.client.stream.EventRead;
+import io.pravega.client.stream.EventStreamReader;
+import io.pravega.client.stream.ReaderConfig;
+import io.pravega.client.stream.ReaderGroupConfig;
+
+import org.apache.flink.api.common.io.GenericInputFormat;
+import org.apache.flink.api.common.io.InputFormat;
+import org.apache.flink.core.io.GenericInputSplit;
+import org.apache.flink.streaming.util.serialization.DeserializationSchema;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Set;
+
+import static io.pravega.connectors.flink.util.FlinkPravegaUtils.createPravegaReader;
+import static io.pravega.connectors.flink.util.FlinkPravegaUtils.generateRandomReaderGroupName;
+
+/**
+ * A Flink {@link InputFormat} that can be added as a source to read from Pravega in a Flink batch job.
+ */
+public class FlinkPravegaInputFormat<T> extends GenericInputFormat<T> {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final long DEFAULT_EVENT_READ_TIMEOUT = 1000;
```

Comment:
This is fine, but on the optimistic side, perhaps a value like 10s would suit better. If this is a batch read and all the stream data is there, then we should in the absence of any glitches never time out.